docs/contributor/draft/rate-limit-design.md

```diff
@@ -0,0 +1,80 @@
+# Rate Limit Proposal
+
+## Spec 
+
+| field                                            | type        | description                                                     | required                      |
+|--------------------------------------------------|-------------|-----------------------------------------------------------------|-------------------------------|
+| wokloadSelector                                  | object      | when skipped, it's applied for the whole namespace workloads, if empty and in istio-system ns, applies to all workloads in all namespaces | no                            |
+| workloadSelector.labels                          | object      | allows to select given workload by a label selector                   | no                            |
+| local                                   | object      | allows to describe local rate limit properties                  | yes                           |
```

Comment:
Should we have it as required? If we introduce global rate limiting it might not be required any longer, so we will have to change this validation once we introduce global rate limiting. If this is fine for us, we need to mention this in the decision and consequences of the ADR.

Reply:
As we discussed, we are ok with that, and let's add it to the ADR.